dotnet/src/Experimental/Process.Core/ProcessBuilder.cs

```diff
@@ -0,0 +1,164 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.SemanticKernel;
+
+/// <summary>
+/// Provides functionality for incrementally defining a process.
+/// </summary>
+public class ProcessBuilder : ProcessStepBuilder
```

Comment:
Is there any plan to subclass `ProcessStepBuilder` anywhere else besides here and `EndStep`?

I do wonder about an alternative pattern where there is a static method / property that provides creates an `EndStep`.  That is...does `EndStep` really need to be a subclass, or can it just be an instance of some _builder_ that takes a certain shape.  As a developer, would I be locked into using `EndStep` or can someone create a custom `EndStep`?

FWIW - `EndStep` feels like it may be "code smell": https://en.wikipedia.org/wiki/Code_smell...but perhaps not a terrible smell. :)

Reply:
I agree that the EndStepis awkward. I have an idea for how to improve this, but for now, I'm keeping it internal (updated) and the public API to trigger the stop is using something like

```
userInputStep
            .OnFunctionResult("GetUserInput")
            .StopProcess();
```